sormas-ui/src/main/java/de/symeda/sormas/ui/samples/SampleCreateForm.java

```diff
@@ -90,6 +95,28 @@ protected void addFields() {
 				false,
 				I18nProperties.getValidationError(Validations.afterDate, testDateField.getCaption(), sampleDateField.getCaption())));
 
+		pathogenTestResultField.addValueChangeListener(e -> {
+			PathogenTestResultType testResult = (PathogenTestResultType) e.getProperty().getValue();
```

Comment:
Move this code to a separate method (setCqValueFieldVisibility with testType and testResult as parameters) and call it in both listeners to avoid duplicate code

Reply:
Done